src/shared/components/selected-species/SpeciesLozenge.tsx

```diff
@@ -56,11 +56,63 @@ const SpeciesLozenge = (props: Props) => {
   return (
     <button className={componentClasses} {...otherProps}>
       <div className={styles.inner}>
-        <span className={styles.name}>{displayName}</span>
-        <span className={styles.assembly}>{props.species.assembly.name}</span>
+        <LozengeContent
+          species={species}
+          displayOption={speciesNameDisplayOption}
+        />
       </div>
     </button>
   );
 };
 
+type DisplayStringProps = {
+  species: CommittedItem;
+  displayOption: SpeciesNameDisplayOption;
+};
+
+const LozengeContent = (props: DisplayStringProps) => {
+  const { species, displayOption } = props;
+  if (displayOption === 'common-name_assembly-name') {
+    return (
+      <>
+        <span className={styles.default}>{species.common_name}</span>
+        <span className={styles.assembly}>{species.assembly.name}</span>
+      </>
+    );
+  } else if (displayOption === 'common-name_type_assembly-name') {
```

Comment:
The styles (as well as the extent of the information) of the type don't look right:

![species-lozenge-info review ensembl org_species-selector](https://github.com/Ensembl/ensembl-client/assets/6834224/1c8405e9-96ab-49a1-beb3-3b0216a4699a)

- compare with XD
- remember that a genome can also be a reference assembly

In my PR, I added dedicated components for different parts of species name. See if they make sense. Perhaps you could reuse them after my PR gets merged (or perhaps use my branch as the base, and rebase your commits on top of my branch).